### ISSUE TITLE
otelcol.connector.spanmetrics  spanmetrics count is growing infinitely  over the period 

### ISSUE THREAD

**Issue by ashish0x02** [grafana/alloy#234]
### What's wrong?

i can see spanmetrics count is growing over the period, once grafana-agent restart then only i can see some drop on `traces_spanmetrics_calls_total`.
i've configured `metrics_flush_interval = "5s"` and i'm using [docker.io/grafana/agent:v0.39.2](http://docker.io/grafana/agent:v0.39.2)
<img width="1978" alt="image" src="https://github.com/grafana/agent/assets/105202331/469fc93f-3da0-4561-875f-7e6951130ec6">
issue is similar to https://github.com/grafana/agent/issues/5271

### Steps to reproduce

config for `otelcol.connector.spanmetrics`
```
// The Spanmetrics Connector will generate RED metrics based on the incoming trace span data.
    otelcol.connector.spanmetrics "tracemetrics" {
        // The namespace explicit adds a prefix to all the generated span metrics names.
        // In this case, we'll ensure they match as closely as possible those generated by Tempo.
        namespace = "traces.spanmetrics"

        // Each extra dimension (metrics label) to be added to the generated metrics from matching span attributes. These
        // need to be defined with a name and optionally a default value (in the following cases, we do not want a default
        // value if the span attribute is not present).
        dimension {
            name = "http.method"
        }
        dimension {
            name = "http.status_code"
        }
        metrics_flush_interval = "5s"

        // A histogram block must be present, either explicitly defining bucket values or via an exponential block.
        // We do the latter here.
        histogram {
            explicit {
              buckets=  ["10ms", "50ms", "100ms", "200ms", "400ms", "1s"]
            }
        }

        // The exemplar block is added to ensure we generate exemplars for traces on relevant metric values.
        exemplars {
            enabled = true
        }

        // Generated metrics data is in OTLP format. We send this data to the OpenTelemetry Prometheus exporter to ensure
        // it gets transformed into Prometheus format data.
        output {
            metrics = [otelcol.processor.filter.default.input]
        }
    }
```

### Software version

agent v0.39.2

**Comment by SatyKrish** [comment]
Could this be a related issue? We observed spanmetrics keep getting produced when there is no new spans being received.

https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/30559

**Comment by rfratto** [comment]
The upstream fix (open-telemetry/opentelemetry-collector-contrib#31106) was brought into Alloy with #711 and released with Alloy v1.1; closing as completed.
